### ISSUE TITLE
Azure DevOps Container Resource Endpoint: Error: credentials key has https[s]:// prefix

### ISSUE THREAD

**Issue by jamjon3** [containers/podman#13691]
/kind bug

**Description**

Using podman-docker on a self-hosted agent running RHEL8, defining a container as a resource with an endpoint (predefined service connection) fails with 

Error: credentials key has https[s]:// prefix

This is due to the way registry service connections are formatted in Azure DevOps which require the leading 'https://' and cannot be omitted. Using podman-docker allows these agents to 'look like docker' to the built in tasks and definition specific for docker (and not podman).

**Steps to reproduce the issue:**

1. Define a service connection that must be prefixed with https://

for the example, we'll call it 'nexus_registry' with the url https://nexus01.example.com and provide valid credentials and save it.
2. Create a yaml pipeline like this and specify the hosted agent pool with RHEL8 VM's with podman and podman-docker setup
```
resources:
  containers:
  - container: somecontainer
    image: nexus01.example.com:5443/containers/base/some_container:latest
    endpoint: nexus_registry
trigger: none
jobs:
  - job: podmanInsideContainerTest
    displayName: Test Podman running inside container
    pool: PodmanOnPremTest
    container: somecontainer
    steps:
      - bash: |
          echo "Test"
```
3. Run the pipeline

**Describe the results you received:**
```
Starting: Initialize containers
/usr/bin/docker version --format '{{.Server.APIVersion}}'
''3.4.2'
Docker daemon API version: ''3.4.2'
/usr/bin/docker version --format '{{.Client.APIVersion}}'
'3.4.2'
Docker client API version: '3.4.2'
/usr/bin/docker ps --all --quiet --no-trunc --filter "label=a77cb7"
/usr/bin/docker network prune --force --filter "label=a77cb7"
/usr/bin/docker login --username "***" --password-stdin ***
Error: credentials key has https[s]:// prefix
##[error]Docker login fail with exit code 125
Finishing: Initialize containers
```
**Describe the results you expected:**

Using regular 'docker' on RHEL7 works fine:
```
WARNING! Your password will be stored unencrypted in /home/DevOps1/.docker/config.json.
Configure a credential helper to remove this warning. See
https://docs.docker.com/engine/reference/commandline/login/#credentials-store

Login Succeeded
/bin/docker pull nexus01.example.com:5443/containers/base/some_container:latest
latest: Pulling from containers/base/some_container
Digest: sha256:7495c6b5669beeca1ca00c873708a4a31007f489373dd1bb183ab994116e4db1
Status: Image is up to date for nexus01.example.com:5443/containers/base/some_container:latest
nexus01.example.com:5443/containers/base/some_container:latest
/bin/docker logout ***
Removing login credentials for nexus01.example.com:5443
/bin/docker info -f "{{range .Plugins.Network}}{{println .}}{{end}}"
bridge
host
ipvlan
macvlan
null
overlay
/bin/docker network create --label 9f0d2f vsts_network_bf643880f9d44c158672e03b444d1ed1
37759e439e563f2d0154091a92a64366a764f4a2d12c06ad8a9eb54e65e6809e
/bin/docker inspect --format="{{index .Config.Labels \"com.azure.dev.pipelines.agent.handler.node.path\"}}" nexus01.example.com:5443/containers/base/some_container:latest
/bin/docker create --name some_container_nexus01examplecom5443containersbasesome_containerlatest_e13130 --label 9f0d2f --network vsts_network_bf643880f9d44c158672e03b444d1ed1  -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/usr/local/share/vsts-agent/160":"/__a/160" -v "/usr/local/share/vsts-agent/_temp":"/__a/_temp" -v "/usr/local/share/vsts-agent/_tasks":"/__a/_tasks" -v "/usr/local/share/vsts-agent/_tool":"/__t" -v "/usr/local/share/vsts-agent/externals":"/__a/externals":ro -v "/usr/local/share/vsts-agent/.taskkey":"/__a/.taskkey" nexus01.example.com:5443/containers/base/some_container:latest "/__a/externals/node/bin/node" -e "setInterval(function(){}, 24 * 60 * 60 * 1000);"
df7827b29b2b44476800af916c7c9fdc610a6ecb3464c1fb0182cdb9a00734f5
/bin/docker start df7827b29b2b44476800af916c7c9fdc610a6ecb3464c1fb0182cdb9a00734f5
df7827b29b2b44476800af916c7c9fdc610a6ecb3464c1fb0182cdb9a00734f5
/bin/docker ps --all --filter id=df7827b29b2b44476800af916c7c9fdc610a6ecb3464c1fb0182cdb9a00734f5 --filter status=running --no-trunc --format "{{.ID}} {{.Status}}"
df7827b29b2b44476800af916c7c9fdc610a6ecb3464c1fb0182cdb9a00734f5 Up Less than a second
/bin/docker exec  df7827b29b2b44476800af916c7c9fdc610a6ecb3464c1fb0182cdb9a00734f5 sh -c "command -v bash"
/usr/bin/bash
whoami 
DevOps1
id -u DevOps1
1000
Try to create a user with UID '1000' inside the container.
/bin/docker exec  df7827b29b2b44476800af916c7c9fdc610a6ecb3464c1fb0182cdb9a00734f5 bash -c "getent passwd 1000 | cut -d: -f1 "
/bin/docker exec  df7827b29b2b44476800af916c7c9fdc610a6ecb3464c1fb0182cdb9a00734f5 useradd -m -u 1000 DevOps1_azpcontainer
Grant user 'DevOps1_azpcontainer' SUDO privilege and allow it run any command without authentication.
/bin/docker exec  df7827b29b2b44476800af916c7c9fdc610a6ecb3464c1fb0182cdb9a00734f5 groupadd azure_pipelines_sudo
/bin/docker exec  df7827b29b2b44476800af916c7c9fdc610a6ecb3464c1fb0182cdb9a00734f5 usermod -a -G azure_pipelines_sudo DevOps1_azpcontainer
/bin/docker exec  df7827b29b2b44476800af916c7c9fdc610a6ecb3464c1fb0182cdb9a00734f5 su -c "echo '%azure_pipelines_sudo ALL=(ALL:ALL) NOPASSWD:ALL' >> /etc/sudoers"
Allow user 'DevOps1_azpcontainer' run any docker command without SUDO.
stat -c %g /var/run/docker.sock
992
/bin/docker exec  df7827b29b2b44476800af916c7c9fdc610a6ecb3464c1fb0182cdb9a00734f5 bash -c "cat /etc/group"
/bin/docker exec  df7827b29b2b44476800af916c7c9fdc610a6ecb3464c1fb0182cdb9a00734f5 groupadd -g 992 azure_pipelines_docker
/bin/docker exec  df7827b29b2b44476800af916c7c9fdc610a6ecb3464c1fb0182cdb9a00734f5 usermod -a -G azure_pipelines_docker DevOps1_azpcontainer
Finishing: Initialize containers
```
**Additional information you deem important (e.g. issue happens only occasionally):**

It would seem there should be something to override this validation perhaps and just 'strip' it if podman finds https:// as the registry specified (aka: make it more 'durable' to masquerade podman as docker with the 'podman-docker' package or something). I'm already using the socket and perhaps in the overlay file that could be set or elsewhere in configuration. I currently have this in my overlay file (this makes the socket available to the docker group which the agent user is a member of... aka: make it work like docker so the agent doesn't know it's actually podman)

# cat /etc/systemd/system/podman.socket.d/override.conf
[Socket]
SocketMode=0660
SocketUser=root
SocketGroup=docker

This sets up the /run folder so it persists on reboot
# cat /etc/tmpfiles.d/podman.conf
d /run/podman 0770 root docker

**Output of `podman version`:**

```
podman version 3.4.2
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.23.1
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.0.32-1.module+el8.5.0+13852+150547f7.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.32, commit: 4b12bce835c3f8acc006a43620dd955a6a73bae0'
  cpus: 8
  distribution:
    distribution: '"rhel"'
    version: "8.5"
  eventLogger: file
  hostname: tpapdvlcibld06
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 4.18.0-348.12.2.el8_5.x86_64
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 11948802048
  memTotal: 16600227840
  ociRuntime:
    name: runc
    package: runc-1.0.3-1.module+el8.5.0+13556+7f055e70.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.0.3
      spec: 1.0.2-dev
      go: go1.16.7
      libseccomp: 2.5.1
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_NET_RAW,CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.8-1.module+el8.5.0+12582+56d94c81.x86_64
    version: |-
      slirp4netns version 1.1.8
      commit: d361001f495417b880f20329121e3aa431a8f90f
      libslirp: 4.4.0
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.1
  swapFree: 34355539968
  swapTotal: 34355539968
  uptime: 17h 21m 34.83s (Approximately 0.71 days)
plugins:
  log:
  - k8s-file
  - none
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /home/DevOps1/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.8-1.module+el8.5.0+13754+92ec836b.x86_64
      Version: |-
        fusermount3 version: 3.2.1
        fuse-overlayfs: version 1.8
        FUSE library version 3.2.1
        using FUSE kernel interface version 7.26
  graphRoot: /home/DevOps1/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 431
  runRoot: /run/user/1000/containers
  volumePath: /home/DevOps1/.local/share/containers/storage/volumes
version:
  APIVersion: 3.4.2
  Built: 1642068949
  BuiltTime: Thu Jan 13 10:15:49 2022
  GitCommit: ""
  GoVersion: go1.16.7
  OsArch: linux/amd64
  Version: 3.4.2
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-3.4.2-9.module+el8.5.0+13852+150547f7.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes - Troubleshooting guide
No - lastest version

**Additional environment details (AWS, VirtualBox, physical, etc.):**
This is a RHEL8 VM running on VMWare off a licensed hypervisor. RHEL7 VM's just running docker work fine (just not RHEL8 with podman)

**Comment by vrothberg** [comment]
Thanks for reaching out!

> @vrothberg I think you fixed this already?

Yes, it has been fixed in the meantime and was backported to 3.4.x.  Since you're reporting against RHEL, please reach out through the Red Hat customer channels (e.g., Bugzilla etc.) when reporting issues on RHEL.

A workaround for `podman login` is to strip the `https://` prefix.

**Comment by jamjon3** [comment]
No. Reverting back to RHEL7 is not what I'm looking for. I'm looking to replace those RHEL7 VM's with RHEL8 and using podman. Currently, podman doesn't work as I described. I have podman docker working and I can build containers with those hosted agents BUT I can't run jobs inside containers using defined resources with a service endpoint.

**Comment by jamjon3** [comment]
I can't 'strip' https:// on a service endpoint in Azure DevOps. Just not an option.

**Comment by vrothberg** [comment]
I double checked.  It will be fixed with RHEL 8.6.

**Comment by jamjon3** [comment]
Thank you @vrothberg ! I'm refusing to install Docker on RHEL8 and getting ready for RHEL9 but I have no option with those service connections defined as 'resources' so that's great news!!!!
